test/e2e/e2e_test.go

```diff
@@ -292,6 +297,140 @@ func pollForConfigMapChange(client *kubernetes.Clientset, compareConfigMap *v1.C
 	})
 }
 
+// GenerateRoute creates a OpenShift routes
+func GenerateRoute(routeClient *routeclient.RouteV1Client, s *v1.Service, namespace string) (*routev1.Route, error) {
+	route, err := routeClient.Routes(namespace).Create(&routev1.Route{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      s.ObjectMeta.Name,
+			Namespace: s.ObjectMeta.Namespace,
+			Labels:    s.ObjectMeta.Labels,
+		},
+		Spec: routev1.RouteSpec{
+			TLS: &routev1.TLSConfig{
+				InsecureEdgeTerminationPolicy: routev1.InsecureEdgeTerminationPolicyRedirect,
+				Termination:                   routev1.TLSTerminationPassthrough,
+			},
+			To: routev1.RouteTargetReference{
+				Kind: "Service",
+				Name: s.ObjectMeta.Name,
+			},
+			Port: &routev1.RoutePort{
+				TargetPort: s.Spec.Ports[0].TargetPort,
+			},
+		},
+	})
+
+	if err = wait.PollImmediate(5*time.Second, 5*time.Minute, func() (done bool, err error) {
+		localRoute, err := routeClient.Routes(namespace).Get(s.ObjectMeta.Name, metav1.GetOptions{})
+		if err != nil && errors.IsNotFound(err) {
+			return false, nil
+		}
+		if !IsAdmitted(localRoute) {
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		return nil, err
+	}
+	return route, nil
+}
+
+func IsAdmitted(route *routev1.Route) bool {
+	for _, ingress := range route.Status.Ingress {
+		if isIngressAdmitted(ingress) {
+			return true
+		}
+	}
+	return false
+}
+
+func isIngressAdmitted(ingress routev1.RouteIngress) bool {
+	admitted := false
+	for _, condition := range ingress.Conditions {
+		if condition.Type == routev1.RouteAdmitted && condition.Status == v1.ConditionTrue {
+			admitted = true
+		}
+	}
+	return admitted
+}
+
+// createService creates a service object and returns a pointer pointing to this object on success.
+func createService(client *kubernetes.Clientset, namespace string, serviceName string, deploymentName string) (*v1.Service, error) {
+	service, err := client.CoreV1().Services(namespace).Create(&v1.Service{
+		TypeMeta: metav1.TypeMeta{},
```

Comment:
Remove this line

Reply:
I have added this line after seeing the similar usage in the `createServingCertAnnotatedService` function in this file.